### DISCUSSION TITLE
Is the perMessageDeflate incorrectly documented ?

### DISCUSSION THREAD

**Discussion by royalpinto** [socketio/socket.io#4520] [Q&A] [answered]
In the [v2 docs](https://socket.io/docs/v2/server-api/) it says the `perMessageDeflate`'s default value is false. But seems like it is not.

```js
const socketio = require('socket.io') // v2
const io = socketio({})
io.listen(3000)
console.log(io.engine.ws.options.perMessageDeflate) // => { threshold: 1024 }
```

In the [v4 docs](https://socket.io/docs/v4/server-options/#permessagedeflate) also it says the default value of it is false.

```js
const socketio = require('socket.io') // v4
const io = socketio({})
io.listen(3000)
console.log(io.engine.ws.options.perMessageDeflate) // undefined
```

But when it is set to true in v4, it becomes same as v2
```js
const socketio = require('socket.io') // v4
const io = socketio({perMessageDeflate: true})
io.listen(3000)
console.log(io.engine.ws.options.perMessageDeflate) // { threshold: 1024 }
```

Mostly in the v2 it is incorrectly documented as false where as in reality is it true ? 🤔

**Comment by darrachequesne** [comment, marked as answer]
Hi! `perMessageDeflate` defaults to `false` starting from `socket.io@2.4.0`: https://github.com/socketio/socket.io/releases/tag/2.4.0

I've updated the v2 documentation to make it clearer: https://socket.io/docs/v2/server-api/
